src/main/java/io/tiledb/java/api/Condition.java

```diff
@@ -0,0 +1,88 @@
+package io.tiledb.java.api;
+
+import io.tiledb.libtiledb.*;
+import java.lang.reflect.Array;
+import java.math.BigInteger;
+
+public class Condition implements AutoCloseable {
+  private Context ctx;
+  private SWIGTYPE_p_tiledb_query_condition_t conditionp;
+  private SWIGTYPE_p_p_tiledb_query_condition_t conditionpp;
+  private Datatype type;
+
+  protected Condition(Context ctx, SWIGTYPE_p_p_tiledb_query_condition_t conditionpp) {
+    this.ctx = ctx;
+    this.conditionp = tiledb.tiledb_query_condition_tpp_value(conditionpp);
+    this.conditionpp = conditionpp;
+  }
+
+  protected Condition(
```

Comment:
We need to handle NULL conditions also. NULLs are passed to TileDB by setting a  value `nullptr` and size of 0. Just a note because I'm not sure below handle this.

Reply:
If I understand correctly you mean that I should handle cases where we add a query condition with size 0 for the value or the whole condition object is null. Right?

Reply:
It seems that the core library does not handle query conditions on nullable attributes yet. 
[https://github.com/TileDB-Inc/TileDB/pull/2141](url)

Reply:
It does support nulls, that's only the first PR.

Reply:
Right! I updated the code to handle null cases. Also added a case in the tests.